pages/ru/snapps/how-to-write-a-snapp.mdx

```diff
@@ -0,0 +1,238 @@
+
+### Install Snapp CLI
+
+#### Dependencies:
+- NodeJS 16+ (or 14 using `node --experimental-wasm-threads`)
+- NPM 6+ 
+- Git 2+ 
+
+<Alert kind="tip">
+
+If you have an older version installed, we suggest installing a newer version using the package manager for your system: <a href="https://brew.sh/">Homebrew</a> (Mac), <a href="https://chocolatey.org/">Chocolatey</a> (Windows), or apt/yum/etc (Linux). On Linux, you may need to install a recent NodeJS version via NodeSource (<a href="https://github.com/nodesource/distributions#debinstall">deb</a> or <a href="https://github.com/nodesource/distributions#rpminstall">rpm</a>), as recommended by the NodeJS Project.
+
+</Alert>
+
+### Start a Project
+
+Now that you have Snapp CLI installed, you can start with an example or start your own project.
+
+#### Option A: Start with an example (recommended)
+
+Examples are based on the standard project structure, but with additional files in the `/src` directory as the only difference.
+
+1. **Install:** Run `snapp example sudoku`. This creates a new project and includes the example files (i.e. the smart contract) inside the project’s `src/` directory. Type `ls` & hit enter to see the files that were created or open the directory in a code editor such as VS Code.
+2. **Run tests:** Run `npm run test`. Tests are written using <a href="https://jestjs.io/">Jest</a>. After running this command, you should see all tests pass. You can also run `npm run testw` to run tests in watch mode, so it will automatically re-run tests when saving changes to your code.
+3. **Build the example:** Run `npm run build`. This will compile your TypeScript into JavaScript inside the project’s `/build` directory.
+4. **Deploy example on Testnet:** (Coming soon.) 
+
+You can view a list of <a href="https://github.com/o1-labs/snapp-cli/tree/main/examples/">all available examples here</a>. 
+
+#### Option B: Start your own project 
+
+1. **Install:** Run `snapp project <myproj>`. Type `ls` and hit enter to see the newly created project structure.
+2. **Run tests:** Run `npm run test`. Tests are written using <a href="https://jestjs.io/">Jest</a>. After running this command, you should see all tests pass. You can also run `npm run testw` to run tests in watch mode, so it will automatically re-run tests when saving changes to your code.
+3. **Build:** Run `npm run build`. This will compile your TypeScript code into JavaScript inside the project’s `/build`. 
+4. **Deploy to Testnet:** (Coming soon.)
+5. **Deploy to Mainnet:** (Coming soon.)
+
+### Writing your smart contract’s custom logic
+
+The goal of this section is to explain the concepts that you will need to understand to write a zero-knowledge-based smart contract.
+
+If you haven’t yet read the <DocLink copy="how snapps work" url="/snapps/how-snapps-work" /> section, please read it first so that this section makes sense.
+
+##### SnarkyJS
```

Comment:
iirc the SnarkyJS section appeared above as well. It might need to be re-organized to remove redundancy.

Reply:
What if we move this section up to follow "Installation" and before "Star a project" section? @jasongitmail